desktop/core/src/desktop/settings.py

```diff
@@ -155,7 +155,7 @@
     'desktop.middleware.SpnegoMiddleware',
     'desktop.middleware.HueRemoteUserMiddleware',
     'django.middleware.locale.LocaleMiddleware',
-    'babeldjango.middleware.LocaleMiddleware',
+    'django_babel.middleware.LocaleMiddleware',
```

Comment:
This will break Python2 as we don't have the lib in desktop/ext-py there?

If compatible we could add it to avoid breaking regular Python 2 or we need to have a `if` using the related middleware depending on the Python version

Reply:
How we update the django_babel in python2? As for python3, we can update the requirement.txt.

Reply:
Nice, almost there!

When we update ext-py, it is a bit different, there is no auto build as these modules are built during the build with `make apps`.

So you would need to do `make apps` again.

Note: often, you need to start with a fresher setup, by doing a `make clean`, `rm -r build`, then `make apps` again.

This is indeed a problem only with Python 2 as with Python 3 we only have requirements.txt.

Reply:
And the doc on how to test the localization https://docs.gethue.com/developer/development/#internationalization

IIRC doing a `make locales` is good enough. Then you can revert the modified files via `git reset --hard`.

Reply:
so I have to choose python 2 for the `make apps` because I have tried with python 3? And the same error appears as above. @romainr 
https://app.circleci.com/pipelines/github/cloudera/hue/4146/workflows/da6bf4c9-c597-4ea2-8ab9-20870dc07a08/jobs/5636

Reply:
Python2 uses libs in desktop/ext-py
Python3 uses libs in requirements.txt

For Python2, I can tweak the CI for you, just I would need a separate PR  where the new lib Django babel is added in ext-py, then will rebuild the image, then we can iterate on the PR removing the old lib and doing all the other changes.

For Python3, the link above is the Python2 run, the correct name is `build-py3.6` https://app.circleci.com/pipelines/github/cloudera/hue/4146/workflows/da6bf4c9-c597-4ea2-8ab9-20870dc07a08/jobs/5637